SR2020/2020-07-05-virtual-competition-timings.md

```diff
@@ -0,0 +1,37 @@
+---
+to: sr2020-teams
+subject: SR2020 Virtual Competition Schedule and New Simulation Version
+---
+
+Hi all
+
+## The Competition Schedule
+
+We're pleased to announce the structure of the SR2020 virtual competition. We will be running 5 sessions over 3 weekends.
+
+You will have until 10am on each day to make changes before your team's code is frozen. We will be in contact soon about our submission process.
+
+**At 3pm we will be live-streaming matches** back-to-back on YouTube.
+
+| Date           | Competition Stage                                                   |
+|----------------|---------------------------------------------------------------------|
+| 11th July 2020 | [League Part 1](https://studentrobotics.org/events/sr2020/virtual-competition-league-1/)       |
+| 12th July 2020 | [League Part 2](https://studentrobotics.org/events/sr2020/virtual-competition-league-2/)       |
+| 18th July 2020 | [League Part 3](https://studentrobotics.org/events/sr2020/virtual-competition-league-3/)       |
+| 19th July 2020 | [League Part 4](https://studentrobotics.org/events/sr2020/virtual-competition-league-4/)       |
+| 25th July 2020 | [Knockouts and Final](https://studentrobotics.org/events/sr2020/virtual-competition-knockouts/)|
+
+## Version 0.3.1 of the Simulation
+
+We have released version 0.3.1 of the competition simulation which you can download from the [Docs page](https://studentrobotics.org/docs/competition-simulator/).
+
+Here are some of the notable changes since 0.2.0:
+
+- Raise an error if you try to `R.sleep()` with negative seconds
```

Comment:
There was actually another change which also errors for `0`, so we probably should include that,
```suggestion
- Raise an error if you try to `R.sleep()` with seconds less than or equal to 0
```

Reply:
Why error for 0? Seems like it should just continue

Reply:
It was part of the PR which fixed an issue where sleeping for small times was rounded to zero and that made it too easy to end up with a stopped simulator. Webots doesn't advance time if you step for zero duration.

Since I actually changed the approach in that PR to round up the sleep time we maybe could now allow zero again, however I hope that won't actually impact anyone.

In any case, disallowing zero is what 0.3.1 does and is worth documenting.